src/eval/operation.rs

```diff
@@ -2513,6 +2531,71 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                     ))
                 }
             }
+            BinaryOp::DictionaryAssume() => {
+                let (
+                    Closure {
+                        body: contract_term,
+                        env: contract_env,
+                    },
+                    _,
+                ) = self.stack.pop_arg(&self.cache).ok_or_else(|| {
+                    EvalError::NotEnoughArgs(3, String::from("dictionary_assume"), pos_op)
+                })?;
+
+                let lbl = match_sharedterm! {t1, with {
+                        Term::Lbl(lbl) => lbl
+                    } else return Err(EvalError::TypeError(
+                        String::from("Lbl"),
+                        String::from("dictionary_assume, 2nd argument"),
+                        fst_pos,
+                        RichTerm {
+                            term: t1,
+                            pos: pos1,
+                        }
+                    ))
+                };
+
+                match_sharedterm! {t2,
+                    with {
+                        Term::Record(record_data) => {
+                            let contract = contract_term.closurize(&mut self.cache, &mut env2, contract_env);
+                            let record_contract = Term::Record(RecordData{
+                                fields: record_data.fields.iter().map(|(key, value)| (*key, Field {
+                                    value: None,
+                                    pending_contracts: vec![PendingContract::new(contract.clone(), lbl.clone())],
+                                    metadata: FieldMetadata {
```

Comment:
Ah, this is a bit tricky. We have to found a "neutral" element for the original field's metadata. I think it might deserve a comment, so that if we extend metadata (for example with custom merge functions or whatnot), we know what we have to do here.

Reply:
In that case it might be better to not use `..Default::default()`. That way we'll be forced by the compiler to consider that question when we change `FieldMetadata` in the future.